File: packages/js-sdk/src/session/out.ts
export enum OutType {
  Stdout = 'Stdout',
  Stderr = 'Stderr',
}

export interface OutResponse {
  type: OutType
  // Unix epoch in nanoseconds
  timestamp: number
  line: string
}

export interface OutStdoutResponse extends OutResponse {
  type: OutType.Stdout
}

export interface OutStderrResponse extends OutResponse {
  type: OutType.Stderr
}

File: packages/python-sdk/e2b_sdk/session/out.py
from enum import Enum
from pydantic import BaseModel

class OutType(str, Enum):
    Stdout = "Stdout"
    Stderr = "Stderr"

class OutResponse(BaseModel):
    type: OutType
    # Unix epoch in nanoseconds
    timestamp: int
    line: str

class OutStdoutResponse(OutResponse):
    type: OutType = OutType.Stdout

class OutStderrResponse(OutResponse):
    type: OutType = OutType.Stderr
